resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -74,7 +72,7 @@
 
     static {
         // Note that we did not use SystemPropertyUtil.getBoolean() here to emulate the behavior of JDK.
-        if ("true".equalsIgnoreCase(SystemPropertyUtil.get("java.net.preferIPv6Addresses"))) {
+        if (Boolean.getBoolean("java.net.preferIPv6Addresses")) {
```

Comment:
IMHO:
* Resolving a String and comparing to "true" is ugly
* Using `SystemPropertyUtil.getBoolean` and evaluate to true when someone was to pass `-Djava.net.preferIPv6Addresses=yes` seems wrong
* There's an idiomatic Java way for this

Of, that's your call. I can revert if you want.